scalajslib/worker/1/src/mill/scalajslib/worker/ScalaJSWorkerImpl.scala

```diff
@@ -202,21 +205,41 @@ class ScalaJSWorkerImpl extends ScalaJSWorkerApi {
 
     val resultFuture = (for {
       (irContainers, _) <- irContainersAndPathsFuture
-      irFiles <- irFileCacheCache.cached(irContainers)
+      irFiles0 <- irFileCacheCache.cached(irContainers)
+      irFiles = if (esModuleMap.isEmpty) {
+        irFiles0
+      } else {
+        if (!minorIsGreaterThanOrEqual(16)) {
+          throw new Exception(
+            s"Remapping EsModule imports will work with scalaJS 1.16 and above. You are using scalaJS ${ScalaJSVersions.current} - consider upgrading?"
+          )
+        }
+        val remapFct = esModuleMap.toSeq.foldLeft((in: String) => in) { case (fct, (s1, s2)) =>
+          val fct2: (String => String) = (in => in.replace(s1, s2))
+          (in => fct(fct2(in)))
+        }
+        irFiles0.map { ImportMappedIRFile.fromIRFile(_)(remapFct) }
+      }
       report <-
         if (useLegacy) {
+          // This uses the legacy linker interface, which is deprecated. The compiler will warn us about it, but the warnings are intentional in a legacy block. Suppress them.
           val jsFileName = "out.js"
           val jsFile = new File(dest, jsFileName).toPath()
+          @annotation.nowarn
           var linkerOutput = LinkerOutput(PathOutputFile(jsFile))
             .withJSFileURI(java.net.URI.create(jsFile.getFileName.toString))
+
           val sourceMapNameOpt = Option.when(sourceMap)(s"${jsFile.getFileName}.map")
           sourceMapNameOpt.foreach { sourceMapName =>
             val sourceMapFile = jsFile.resolveSibling(sourceMapName)
+            @annotation.nowarn
+            val outFct = PathOutputFile(sourceMapFile)
             linkerOutput = linkerOutput
-              .withSourceMap(PathOutputFile(sourceMapFile))
+              .withSourceMap(outFct)
               .withSourceMapURI(java.net.URI.create(sourceMapFile.getFileName.toString))
           }
-          linker.link(irFiles, moduleInitializers, linkerOutput, logger).map {
+          @annotation.nowarn
```

Comment:
Can you make first a cleanup PR with these changes so we merge that first, and then we can discuss only the change related to the new feature?